### ISSUE TITLE
Exception with Angular-cli 6.0.1

### ISSUE THREAD

**Issue by pacoita** [angular/angularfire2#1638]
I updated my application to angular 6 (using angular-cli), but when I try to build it using `ng build` I get the following error:

`ERROR in node_modules/angularfire2/firebase.app.module.d.ts(10,5): error TS2416: Property 'messaging' in type 'FirebaseApp' is not assignable to the same property in base type 'App'.
  Type '() => Messaging' is not assignable to type 'MessagingFactory'.
    Property 'isSupported' is missing in type '() => Messaging'.`

#### angularfire2: 5.0.0-rc.8.1-next

**Package.json**
```
  "dependencies": {
    "@agm/core": "^1.0.0-beta.1",
    "@angular/animations": "^6.0.1",
    "@angular/cdk": "^6.0.1",
    "@angular/common": "^6.0.1",
    "@angular/compiler": "^6.0.1",
    "@angular/core": "^6.0.1",
    "@angular/flex-layout": "^6.0.0-beta.15",
    "@angular/forms": "^6.0.1",
    "@angular/http": "^6.0.1",
    "@angular/material": "^6.0.1",
    "@angular/platform-browser": "^6.0.1",
    "@angular/platform-browser-dynamic": "^6.0.1",
    "@angular/platform-server": "^6.0.1",
    "@angular/router": "^6.0.1",
    "@angular/service-worker": "^6.0.1",
    "angularfire2": "^5.0.0-rc.8.1-next",
    "codelyzer": "^4.3.0",
    "core-js": "^2.5.3",
    "firebase": "^5.0.2",
    "hammerjs": "^2.0.8",
    "rxjs": "^6.1.0",
    "rxjs-compat": "^6.1.0",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.6.1",
    "@angular/cli": "^6.0.1",
    "@angular/compiler-cli": "^6.0.1",
    "@types/googlemaps": "^3.30.8",
    "@types/hammerjs": "^2.0.34",
    "@types/jasmine": "2.5.46",
    "@types/node": "~7.0.18",
    "http-server": "^0.11.1",
    "jasmine-core": "~2.6.1",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "node-sass": "^4.9.0",
    "protractor": "~5.1.0",
    "source-map-explorer": "^1.3.3",
    "tslint": "^5.10.0",
    "typescript": "^2.7.2"
  }
}
```

**Comment by jamesdaniels** [comment]
Seems like a mismatch of libs vs. typings...

**Comment by rjcoupe** [comment]
Same happening here, identical versions of Angular, Angularfire2 and Firebase.

**Comment by jamesdaniels** [comment]
You'll need to ensure that your `@firebase/*` and `@firebase/*-types` are on the correct versions. If you continue to have troubles please file a ticket with the Firebase JS SDK https://github.com/firebase/firebase-js-sdk

**Comment by SergioV85** [comment]
I have the same issue.

Package.json
```
  "dependencies": {
    "@angular/animations": "~6.0.1",
    "@angular/cdk": "~6.0.1",
    "@angular/common": "~6.0.1",
    "@angular/compiler": "~6.0.1",
    "@angular/core": "~6.0.1",
    "@angular/forms": "~6.0.1",
    "@angular/http": "~6.0.1",
    "@angular/platform-browser": "~6.0.1",
    "@angular/platform-browser-dynamic": "~6.0.1",
    "@angular/platform-server": "~6.0.1",
    "@angular/router": "~6.0.1",
    "@ngrx/core": "^1.2.0",
    "@ngrx/effects": "~5.2.0",
    "@ngrx/store": "~5.2.0",
    "@ngrx/store-devtools": "~5.2.0",
    "@nguniversal/module-map-ngfactory-loader": "~6.0.0",
    "angularfire2": "^5.0.0-rc.8.1-next",
    "bootstrap": "~4.1.0",
    "cookie-parser": "^1.4.3",
    "core-js": "^2.5.1",
    "firebase": "~5.0.2",
    "font-awesome": "^4.7.0",
    "moment": "^2.22.1",
    "ng4-social-login": "^1.3.5",
    "ngx-bootstrap": "^2.0.1",
    "ngx-chips": "^1.9.2",
    "ramda": "^0.25.0",
    "rxjs": "~6.1.0",
    "rxjs-compat": "^6.1.0",
    "ts-loader": "^3.1.1",
    "xhr2": "^0.1.4",
    "zone.js": "^0.8.18"
  },
  "devDependencies": {
    "@angular/cli": "~6.0.1",
    "@angular/compiler-cli": "~6.0.1",
    "@angular/language-service": "~6.0.1",
    "@firebase/app-types": "^0.3.1",
    "@types/jasmine": "~2.8.3",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~8.0.53",
    "@types/ramda": "0.25.20",
    "codelyzer": "~4.3.0",
    "jasmine-core": "~2.8.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.7.2",
    "webpack-bundle-analyzer": "^2.9.0",
    "@angular-devkit/build-angular": "~0.6.1"
  }
```

**Comment by pacoita** [comment]
@SergioV85 a new release has been published (`5.0.0-rc.8.0`). 
By updating to this version, instead of using `5.0.0-rc.8.1-next`, I fixed the error in my solution, therefore I would suggest you to give it a try.

Now I have only a warning:
`83:36-54 "export 'firestore' (imported as 'firebase') was not found in 'firebase/app'`

But it will be easier to fix.

Thanks @jamesdaniels!
